### PR TITLE
kpromo: Add `krel promote-images` functionality

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

From https://github.com/kubernetes/release/tree/8742f143657ad3be35ad6c044e74e87f91cf39dc.

Slack convo: https://kubernetes.slack.com/archives/C2C40FMNF/p1636407926086000?thread_ts=1635789741.040200&cid=C2C40FMNF

- kpromo: Add krel promote-images functionality
- kpromo(pr): Remove dependency on k8s.io/release
- kpromo(pr): Enable as subcommand of `kpromo`
- kpromo(pr): Move image promoter constructs to separate package
- kpromo: Build v3.3.0-beta.2-1 image

Signed-off-by: Stephen Augustus <foo@auggie.dev>
Co-authored-by: Adolfo García Veytia (Puerco) <adolfo.garcia@uservers.net>
Co-authored-by: Sascha Grunert <sgrunert@redhat.com>
Co-authored-by: Carlos Panato <ctadeu@gmail.com>
Co-authored-by: Cecile Robert-Michon <cerobert@microsoft.com>
Co-authored-by: Nabarun Pal <pal.nabarun95@gmail.com>

FYI to the previous contributors of this code: @puerco @saschagrunert @cpanato @CecileRobertMichon @palnabarun 

#### Special notes for your reviewer:

~Still fixing this up locally, since it was more involved than I initially calculated, so expect test failures.~

#### Does this PR introduce a user-facing change?

```release-note
- kpromo: Add krel promote-images functionality
- kpromo(pr): Remove dependency on k8s.io/release
- kpromo(pr): Enable as subcommand of `kpromo`
- kpromo(pr): Move image promoter constructs to separate package
- kpromo: Build v3.3.0-beta.2-1 image
```

### CODE REVIEW

File: cmd/kpromo/cmd/pr/pr.go [hunk 1]
```diff
@@ -0,0 +1,358 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (justaugustus) [inline cmd/kpromo/cmd/pr/pr.go:2]: Copyright date was purposely left intact as 2020.

**Comment by justaugustus** [comment]
There are some other bits I can think of fixing (like merging the functionality of and deprecating `cip-mm`), but I don't want to boil the ocean...

So, ready for review:
/assign @puerco @Verolop 
cc: @kubernetes-sigs/release-engineering

File: cmd/kpromo/cmd/pr/pr.go [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+		args += " --reviewers" + opts.reviewers
+	}
+
+	for _, tag := range opts.tags {
+		args += " --tag " + tag
+	}
+
+	prBody := fmt.Sprintf("Image promotion for %s %s\n", opts.project, strings.Join(opts.tags, " / "))
+	prBody += "This is an automated PR generated from `kpromo`\n"
+	prBody += fmt.Sprintf("```\nkpromo pr %s\n```\n\n", args)
```
Comment (puerco) [inline cmd/kpromo/cmd/pr/pr.go:352]: Nice detail capturing the command here

Comment (justaugustus) [reply cmd/kpromo/cmd/pr/pr.go:352]: > Nice detail capturing the command here

Heh, I can't take credit!
That was @palnabarun in https://github.com/kubernetes/release/pull/2320 and @CecileRobertMichon in https://github.com/kubernetes/release/pull/2280. :)

File: cmd/kpromo/cmd/pr/pr.go [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+		}
+	}
+
+	// add the modified manifest to staging
+	logrus.Debugf("Adding %s to staging area", imagesListPath)
+	if err := repo.Add(imagesListPath); err != nil {
+		return errors.Wrap(err, "adding image manifest to staging area")
+	}
+
+	commitMessage := "releng: Image promotion for " + opts.project + " " + strings.Join(opts.tags, " / ")
```
Comment (CecileRobertMichon) [inline cmd/kpromo/cmd/pr/pr.go:271]: small nit I had from using this recently: do you think we could get rid of the `releng:` prefix? or maybe just add it when it's for the Kubernetes project? Not sure if it's relevant for other projects

Comment (justaugustus) [reply cmd/kpromo/cmd/pr/pr.go:271]: Mind if I fix in a follow-up?
There are more things I'd like to fix about the message, but they're a little out-of-scope for this PR.

Comment (CecileRobertMichon) [reply cmd/kpromo/cmd/pr/pr.go:271]: sounds good 👌

**Comment by CecileRobertMichon** [comment]
lgtm

for the non-Kubernetes project use case

File: cmd/kpromo/cmd/pr/pr.go (hunk 3 above)
Comment (justaugustus) [reply cmd/kpromo/cmd/pr/pr.go:271]: > small nit I had from using this recently: do you think we could get rid of the `releng:` prefix? or maybe just add it when it's for the Kubernetes project? Not sure if it's relevant for other projects

Will be fixed in https://github.com/kubernetes-sigs/promo-tools/pull/460.

**Comment by puerco** [comment]
Awesome, let the promo-tools hackathon begin !
/lgtm

**Pull request merged**